gsa/src/web/pages/scanners/dialog.js

```diff
@@ -140,211 +165,233 @@ class ScannerDialog extends React.Component {
-        GMP_SCANNER_TYPE,
-        OSP_SCANNER_TYPE,
-        GREENBONE_SENSOR_SCANNER_TYPE,
-      ];
-    } else {
-      SCANNER_TYPES = [GMP_SCANNER_TYPE, OSP_SCANNER_TYPE];
-    }
+  let {credential_id} = props;
 
-    let {credential_id} = this.props;
-
-    const data = {
-      ca_pub,
-      comment,
-      host,
-      id,
-      name,
-      port,
-      which_cert,
-    };
-
-    const scannerTypesOptions = map(SCANNER_TYPES, scannerType => ({
-      label: scannerTypeName(scannerType),
-      value: scannerType,
-    }));
-
-    const scanner_credentials = filter_credentials(credentials, type);
-    const is_edit = isDefined(scanner);
-    const isInUse = isDefined(scanner) && scanner.isInUse();
-    const show_cred_info =
-      isDefined(scanner) &&
-      isDefined(scanner.credential) &&
-      scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
-
-    const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
-    const isOspScannerType = type === OSP_SCANNER_TYPE;
-
-    if (isGreenboneSensorType) {
-      credential_id = '';
-    }
+  const scannerTypesOptions = map(SCANNER_TYPES, scannerType => ({
+    label: scannerTypeName(scannerType),
+    value: scannerType,
+  }));
+
+  const scanner_credentials = filter_credentials(credentials, type);
+  const is_edit = isDefined(scanner);
+  const isInUse = isDefined(scanner) && scanner.isInUse();
+  const show_cred_info =
+    isDefined(scanner) &&
+    isDefined(scanner.credential) &&
+    scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
+
+  const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
+  const isOspScannerType = type === OSP_SCANNER_TYPE;
+
+  if (isGreenboneSensorType) {
+    credential_id = '';
+  }
+
+  const [activeTab, setActiveTab] = useState(0);
+
+  const totalPages = 2;
+
+  const {page, handlePageTurn, prevDisabled, nextDisabled} = usePageTurn({
+    activeTab,
+    lastPage: totalPages - 1,
+  });
+
+  useEffect(() => {
+    setActiveTab(page);
+  }, [page]);
+
+  return (
+    <SaveDialog
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      multiStep={true} // the option to switch to the MultiStepFooter. Leaving it out defaults to false
+      onNext={() => handlePageTurn('Next')} // 'Next' turns the next page
+      prevDisabled={prevDisabled}
+      nextDisabled={nextDisabled}
+      onPrev={() => handlePageTurn('Previous')} // 'Previous' goes back
+      defaultValues={data}
+      values={{
+        credential_id,
+        type,
+      }}
+    >
+      {({values: state, onValueChange}) => {
+        return (
+          <Layout flex="column">
+            <DialogTabLayout grow="1" align={['start', 'end']}>
+              <TabList
+                active={activeTab}
+                align={['start', 'stretch']}
+                onActivateTab={handleActivateTab}
+              >
+                <Tab style={{backgroundColor: 'red'}}>{_('Metadata')}</Tab>
```

Comment:
Please use styled-components to style components instead of the style attribute. Also, I don't get, why this needs to be red? Also also: if possible, please use colors from our theme.js instead of defining new ones.

Reply:
This is code I left in by mistake. I will get rid of it.